embassy-net/src/lib.rs

```diff
@@ -208,6 +230,59 @@ impl<D: Driver + 'static> Stack<D> {
         .await;
         unreachable!()
     }
+
+    #[cfg(feature = "dns")]
+    async fn dns_query(
```

Comment:
why not make this `pub`? The only reason you'd want to go through `DnsSocket` is if you need embedded-nal, otherwise directly calling this would be less boilerplate.